src/crud/useFetchMediaUrl.ts

```diff
@@ -0,0 +1,158 @@
+import { useEffect, useReducer, useRef } from 'react';
+import { useGlobal } from 'reactn';
+import Axios from 'axios';
+import { API_CONFIG, isElectron } from '../api-variable';
+import Auth from '../auth/Auth';
+import { remoteIdGuid, remoteId } from '../crud';
+import { dataPath, PathType } from '../utils/dataPath';
+import { MediaFile } from '../model';
+import { infoMsg, logError, Severity } from '../utils';
+// See: https://www.smashingmagazine.com/2020/07/custom-react-hook-fetch-cache-data/
+
+export enum MediaSt {
+  'IDLE',
+  'PENDING',
+  'FETCHED',
+  'SELECTED',
+  'ERROR',
+}
+
+export interface IMediaState {
+  status: MediaSt;
+  error: null | string;
+  url: string; // temporary url
+  urlMediaId: string; // media id
+  trackedTask: string; // passage id
+}
+export const mediaClean: IMediaState = {
+  status: MediaSt.IDLE,
+  error: null,
+  url: '',
+  urlMediaId: '',
+  trackedTask: '',
+};
+
+type Action =
+  | { type: MediaSt.PENDING; payload: string } // mediaId
+  | { type: MediaSt.FETCHED; payload: string } // temporary url
+  | { type: MediaSt.SELECTED; payload: string } // passageId
+  | { type: MediaSt.ERROR; payload: string };
+
+const stateReducer = (state: IMediaState, action: Action): IMediaState => {
+  switch (action.type) {
+    case MediaSt.PENDING:
+      return {
+        ...mediaClean,
+        status: MediaSt.PENDING,
+        urlMediaId: action.payload,
+      };
+    case MediaSt.FETCHED:
+      return { ...state, status: MediaSt.FETCHED, url: action.payload };
+    case MediaSt.SELECTED:
+      return { ...state, status: MediaSt.FETCHED, trackedTask: action.payload };
```

Comment:
Were you going to remove the SELECTED from here?  If not...I don't think the status should be FETCHED if you're changing tasks...presumably this will kick off a new fetch somewhere so maybe it doesn't matter.